File: laboratorios/lab02/l-java/src/main/java/com/tuorg/poo/clases/Universidad.java
package com.tuorg.poo.clases;
import java.util.ArrayList;

public class Universidad {
    String nombre;
    // Agregacion: Una universidad tiene muchos cursos
    ArrayList<Curso> cursos;

    public Universidad(String nombre) {
        this.nombre = nombre;
        this.cursos = new ArrayList<>();
    }

    public String getNombre() {
        return nombre;
    }

    public void agregarCursos(Curso curso) {
        this.cursos.add(curso);
    }

    public ArrayList<Curso> getCursos() {
        return cursos;
    }
}

File: laboratorios/lab02/l-javascript/clases/Universidad.js
import { Curso } from "./Curso.js"

export class Universidad {
  #nombre
  cursos = []

  constructor(nombre) {
    this.#nombre = nombre
  }

  get nombre() {
    return this.#nombre
  }
  set nombre(nombre) {
    this.#nombre = nombre
  }

  agregarCursos(curso) {
    if (curso instanceof Curso) {
      this.cursos.push(curso)
    }
  }
  get cursos() {
    return this.cursos
  }

  toString() {
    return `Universidad{nombre='${this.#nombre}', cursos=${this.cursos}}`
  }
}